src/components/editor/EditorLogin.jsx

```diff
@@ -51,6 +47,42 @@ export default class Login extends BaseComponent {
     }
   }
 
+  handleRedirect() {
+    let url = this.props.location.query.url || '';
+    if (url) {
+      _.forEach(this.props.location.query, (q, name) => {
+        if (name !== 'url') {
+          url = `${url}&${name}=${q}`;
+        }
+      });
+    }
+    browserHistory.push(url);
+  }
+
+  onGoogleResponse(response) {
+    if (response.error) {
+      alert(`Google Login Failed.\n${response.error}`);
+      return;
+    }
+
+    const token = response.tokenId;
+
+    const xhr = new XMLHttpRequest();
+    xhr.open('POST', '/googlelogin', true);
+    xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
+
+    const thisComponent = this;
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+        thisComponent.handleRedirect();
```

Comment:
I saw your question asking for help here before, so don't worry I know you're aware this is awkward haha!

Have you tried just writing `this.handleRedirect()` here though? Because one of the features of arrow functions is exactly that they bind `this` to be the `this` from the context they are created in, so I'd be very surprised if it didn't work.

Reply:
oops I misunderstood arrow functions after all.. you're right, I thought arrow functions just bind `this` to the current scope